lib/grape/dsl/routing.rb

```diff
@@ -77,9 +77,12 @@ def do_not_route_options!
           namespace_inheritable(:do_not_route_options, true)
         end
 
-        def mount(mounts)
+        def mount(mounts, with: {})
           mounts = { mounts => '/' } unless mounts.respond_to?(:each_pair)
           mounts.each_pair do |app, path|
+            if app.respond_to?(:mount_instance)
+              return mount(app.mount_instance(configuration: with) => path)
```

Comment:
This return seems odd, can you explain? What if we have more `mounts`? I think it should be a `if ... else` for whatever follows?

Reply:
So the point of this return is not to execute the rest of the code as part of a Grape::API, only as part of a Grape::API::Instance. However, I do agree that return is the wrong approach, I just need to call next.
I'm surprised no test complained. Perhaps we don't have test coverage for mounting multiple apps, path on a single call?
Can you help me write a failing spec to make sure I better understand what the use case of this `each_pair`?

Do we support something like:
```
mount(API1 => '/api_1', API2 => '/api_2'})
```
?

Reply:
We're definitely missing tests.

I didn't know that we support this kind of mounting, but it looks like! Thanks for looking into it.